agent/config.go

```diff
@@ -2121,6 +2123,99 @@ func ReadConfigPaths(paths []string) (*Config, error) {
 	return result, nil
 }
 
+// ResolveTmplAddrs iterates over the myriad of addresses in the agent's config
+// and performs go-sockaddr/template Parse on each known address in case the
+// user specified a template config for any of their values.
+func (c *Config) ResolveTmplAddrs() (err error) {
+	parse := func(addr *string, name string) {
+		if *addr == "" || err != nil {
+			return
+		}
+		var ip string
+		ip, err = parseSingleIPTemplate(*addr)
+		if err != nil {
+			err = fmt.Errorf("Resolution of %s failed: %v", name, err)
+			return
+		}
+		if strings.HasPrefix(ip, "unix://") {
```

Comment:
I tested both master branch and this one passing a unix socket to advertise_addr. It actually failed even before it gets to this method, it fails at [VerifyUniqueListeners](https://github.com/hashicorp/consul/blob/496b0bcf07ad2a6610b1fa3f2db3de00ef9c4eec/agent/config.go#L1026-L1026). So we should be fine as is.